src/bin/hive_chain.rs

```diff
@@ -36,43 +37,75 @@ impl Decoder for BlockFileCodec {
     }
 }
 
+/// The inputs to the binary.
+#[derive(Parser, Debug)]
+pub struct Args {
+    /// The path to the chain file for the hive test.
+    #[clap(short, long)]
+    chain_path: PathBuf,
+    /// The relayer address.
+    #[clap(long)]
+    relayer_address: Felt,
+    /// The relayer private key.
+    #[clap(long)]
+    relayer_pk: Felt,
+}
+
+const STARKNET_RPC_URL: &str = "http://0.0.0.0:5050";
+const MAX_FELTS_IN_CALLDATA: &str = "30000";
+
 /// Inspired by the Import command from Reth.
 /// https://github.com/paradigmxyz/reth/blob/main/bin/reth/src/commands/import.rs
 #[tokio::main]
 async fn main() -> eyre::Result<()> {
-    let chain_path = Path::new(&std::env::var("CHAIN_PATH")?).to_path_buf();
-    let relayer_address = Felt::from_str(&std::env::var("RELAYER_ADDRESS")?)?;
+    let args = Args::parse();
+
+    // Get the provider
+    let provider = JsonRpcClient::new(HttpTransport::new(Url::from_str(STARKNET_RPC_URL)?));
+    let starknet_provider = StarknetProvider::new(provider);
+
+    // Set the env
+    std::env::set_var("RELAYER_PRIVATE_KEY", format!("0x{:x}", args.relayer_pk));
+    std::env::set_var("MAX_FELTS_IN_CALLDATA", MAX_FELTS_IN_CALLDATA);
+    std::env::set_var("STARKNET_NETWORK", STARKNET_RPC_URL);
+
+    // Set the chain id
+    let chain_id = starknet_provider.chain_id().await?;
+    let modulo = 1u64 << 53;
+    let chain_id_mod: u64 = (Felt::from(modulo).to_bigint() & chain_id.to_bigint()).try_into()?;
+    let _ = CHAIN_ID.get_or_init(|| Felt::from(chain_id_mod));
```

Comment:
Here I would say that we could simplify to a simple modulo operation like:


```rust
    let _ = CHAIN_ID.get_or_init(|| chain_id.div_rem(Felt::from(modulo)).1);
```

but if this is too much for this PR let us create a follow up issue, wdyt?

Reply:
https://github.com/kkrt-labs/kakarot-rpc/issues/1408 done!